### DISCUSSION TITLE
stand-alone DIRAC:: how to automatically point to configuration

### DISCUSSION THREAD

**Discussion by adriansev** [DIRACGrid/DIRAC#6651] [Q&A] [answered]
Hi! trying to use DIRAC stand-alone (without DIRACOS2 - this is on f36, soon to be f37) i hit 2 problems:
1. automatically point to a generated configuration (i cannot replicate all commands to add --cfg=myconf_file)
2. dirac-configure try to download certs and vomsdir in /usr (without any possibility to change the location)

Are there any solutions to these?
Thanks a lot!

**Comment by fstagni** [comment]
Hello, 
can you share a few info?
- which DIRAC version you are using
- which (server) setup you are targeting
- the procedure you followed

There is already an open issue: https://github.com/DIRACGrid/DIRAC/issues/6496 and a discussion on this topic: https://github.com/DIRACGrid/DIRAC/discussions/6121


PS: I guess that "f36" is Fedora 36...?

**Reply by adriansev** [reply]
@fstagni 

> Hello, can you share a few info?

Hi! and thank you for looking at this :)

> * which DIRAC version you are using
```
(base) ALIBUILD##[Thursday 05.01.23 18:32 adrian@sev] ~ $
dirac-version
8.0.8
```

> * which (server) setup you are targeting

just the most simple and plain client

> * the procedure you followed

this one: https://dirac.readthedocs.io/en/latest/UserGuide/GettingStarted/InstallingClient/index.html

but the configure step was:
`dirac-configure -ddd -F -V auger -C dips://dirac-conf.in2p3.fr:9135/Configuration/Server -I`

and the generated configuration look like this:
https://codimd.web.cern.ch/s/SxHtI51in

> 
> There is already an open issue: #6496 and a discussion on this topic: #6121
> 
> PS: I guess that "f36" is Fedora 36...?

yes, Fedora 36..

Partly i solved problem 2 by enabling DIRACOS, but doing `dirac-proxy-init` i still get:
`Cannot get URL for Framework/ProxyManager in setup Test: RuntimeError('Option /DIRAC/Setups/Test/Framework is not defined')
`

and i get this:
```
(base) ALIBUILD##[Thursday 05.01.23 18:41 adrian@sev] ~/bin (master) $
dirac-proxy-get-uploaded-info -ddd
2023-01-05 16:41:33 UTC Framework [139894984496960] DEBUG: dirac.cfg should be at /home.hdd/adrian/DIRACOS/etc/dirac.cfg
2023-01-05 16:41:33 UTC Framework [139894984496960] DEBUG: CFG merged
2023-01-05 16:41:33 UTC Framework [139894984496960] DEBUG: Updating configuration internals
2023-01-05 16:41:33 UTC Framework [139894984496960] DEBUG: Updating configuration internals
Parsing command line
Trying to load Resources.LogBackends.StdoutBackend
Trying to load COMDIRAC.Resources.LogBackends.StdoutBackend
Trying to load DIRAC.Resources.LogBackends.StdoutBackend
2023-01-05 16:41:33 UTC dirac-proxy-get-uploaded-info [139894984496960] FATAL: Missing mandatory local configuration option /DIRAC/Setup
2023-01-05 16:41:33 UTC dirac-proxy-get-uploaded-info [139894984496960] ERROR: There were errors when loading configuration
```

Thanks a lot!

**Reply by fstagni** [reply]
`setup Test` is most probably the issue here. In general there's nothing "wrong" in what you are doing, so as the client gets installed correctly, just it's not configured correctly. When you issue `dirac-configure`, you are most probably not presented with the correct options for your case. For this I defer to @atsareg who's managing the installation at IN2P3.

**Comment by adriansev** [comment]
So, wouldn't be nice to start another discussion on the same subject, so i will continue here:
while i solved point 2. by copying the content of relevant voms dir and vomses directories and exported the needed X509 vars, the main problem (point 1.) remains:
1. how to automatically point to a generated configuration (i cannot replicate all commands to add --cfg=myconf_file)

Is there (or can be added) an env var (maybe something like DIRAC_CONF) that would have the same effect like `--cfg=myconf_file` ?

Thanks a lot!!

**Reply by andresailer** [reply]
DIRACSYSCONFIG should do that:
https://dirac.readthedocs.io/en/latest/AdministratorGuide/ServerInstallations/environment_variable_configuration.html?highlight=Environment%20variables#environment-variables-to-configure-dirac-behaviour

**Reply by adriansev** [reply]
sorry for the noise, i also just fount it in src/DIRAC/ConfigurationSystem/Client/LocalConfiguration.py:454
and it works as intended.
Thanks a lot!

**Comment by adriansev** [comment, marked as answer]
the answer is given by @andresailer in https://github.com/DIRACGrid/DIRAC/discussions/6651#discussioncomment-5087849
